public/js/components/FormFields/FormFieldArrayWrapper.js

```diff
@@ -18,27 +18,51 @@ export default class FormFieldArrayWrapper extends React.Component {
     ])
   }
 
+  //We store new items in state to avoid sending invalid items back to the server before they're ready
+  state = {
+    newItems: []
+  }
+
   onAddClick = () => {
-    const existingValues = this.props.fieldValue || [];
-    this.props.onUpdateField(existingValues.concat([undefined]));
+    this.setState({
+      newItems: this.state.newItems.concat([undefined])
+    });
   }
 
   renderValue(value, i) {
 
     const updateFn = (newValue) => {
-      //Find the value in the array to change
-      const newFieldValue = this.props.fieldValue.map((oldValue) => {
-        return value === oldValue ? newValue : oldValue;
-      });
-      this.props.onUpdateField(newFieldValue);
+      if (this.state.newItems.indexOf(value) > -1) {
+        //It's the first update to a new item - add it to props and remove from newItems
+        this.setState({
+          newItems: this.state.newItems.length > 1 ? this.state.newItems.slice(1) : []
+        });
+
+        const update = this.props.fieldValue ? this.props.fieldValue.concat([newValue]) : [newValue];
+        this.props.onUpdateField(update);
+      } else {
+        //Find the value in the array to change
+        const newFieldValue = this.props.fieldValue.map((oldValue) => {
+          return value === oldValue ? newValue : oldValue;
+        });
+
+        this.props.onUpdateField(newFieldValue);
+      }
     };
 
     const removeFn = (removeIndex) => {
-      const newFieldValue = this.props.fieldValue.filter((value, currentIndex) => {
-        return currentIndex !== removeIndex;
-      });
+      if (this.props.fieldValue && this.props.fieldValue.length > removeIndex) {
```

Comment:
`_.get('this.props', 'fieldValue[removeIndex]')` _should_ work

Reply:
I'm not sure why, but I can't get this to work :(

Reply:
`_.get('this.props.fieldValue', 'length', -1) > removedIndex` ?